KalturaWowzaServer/src/main/java/com/kaltura/media_server/modules/LiveStreamSettingsModule.java

```diff
@@ -0,0 +1,308 @@
+
+		public PacketListener() {
+			this.lastPacketTimeCode = 0;
+			this.basePacketTimeCode = 0;
+			this.baseSystemTime = 0;
+		}
+
+		public void onLivePacket(IMediaStream stream, AMFPacket thisPacket) {
+
+			long currentTime = System.currentTimeMillis();
+			long currentPacketTimeCode = thisPacket.getAbsTimecode();
+			long currentPTS = thisPacket.getTimecode();
+			boolean isAudio = thisPacket.isAudio();
+			boolean isVideo = thisPacket.isVideo();
+			String streamName = stream.getName();
+
+			if (this.entryId == null) {
+				this.entryId = extractEntryId(streamName);
+				registerPacketListener(streamName, this);
+			}
+
+			// get global entry baseSystemTime
+			long liveEntryBaseTimestamp = getLiveEntryBaseSystemTime(this.entryId, currentTime);
+
+			logger.debug("Stream name: " + streamName + " currentTime: " + currentTime + " current TimeCode: " + currentPacketTimeCode + " currentPTS " + currentPTS + " Audio " + Boolean.toString(isAudio) + " Video " + Boolean.toString(isVideo));
+
+			if (liveEntryBaseTimestamp != this.baseSystemTime ||
+					this.baseSystemTime == 0 || this.lastPacketTimeCode == 0
+					|| Math.abs(currentPacketTimeCode - this.lastPacketTimeCode) > maxAllowedPTSDriftMillisec) {
+
+				if (this.baseSystemTime == liveEntryBaseTimestamp && currentTime != this.baseSystemTime) {
+
+					// update globe entry base timestamp
+					liveEntryBaseTimestamp = setLiveEntryBaseSystemTime(this.entryId, currentTime);
+				}
+				this.basePacketTimeCode = currentPacketTimeCode;
+				this.baseSystemTime = liveEntryBaseTimestamp;
+			}
+			this.lastPacketTimeCode = currentPacketTimeCode;
+
+			long packetTimeCode = currentPacketTimeCode - this.basePacketTimeCode + this.baseSystemTime;
+			thisPacket.setAbsTimecode(packetTimeCode);
+		}
+
+	}
+
+	public long getLiveEntryBaseSystemTime(String entryId, long baseSystemTime) {
+		try {
+			if (!this.mapLiveEntryToBaseSystemTime.containsKey(entryId)) {
+				this.mapLiveEntryToBaseSystemTime.put(entryId, baseSystemTime);
+			} else {
+				baseSystemTime = this.mapLiveEntryToBaseSystemTime.get(entryId);
+			}
+		} catch (Exception e) {
+			if (Boolean.valueOf(entryId)) {
+				logger.error("(" + entryId + ") fail to get base timestamp for live entry." + e.toString());
+			} else {
+				logger.error("( UNDEFIMED ENTRY ID ) could not get base timestamp, entryId is invalid. " + e.toString());
+			}
+		}
+
+		return baseSystemTime;
+	}
+
+	public long setLiveEntryBaseSystemTime(String entryId, long baseSystemTime) {
+		this.mapLiveEntryToBaseSystemTime.put(entryId, baseSystemTime);
+		return baseSystemTime;
+	}
+
+	public void incrementStreams(String streamName) {
+		String entryId = extractEntryId(streamName);
+		if (mapLiveEntryStreamsCount.containsKey(entryId)) {
+			mapLiveEntryStreamsCount.put(streamName, mapLiveEntryStreamsCount.get(entryId) + 1);
+		} else {
+			mapLiveEntryStreamsCount.put(streamName, new Long(1));
+		}
+	}
+
+	public void decrementStreams(String streamName) {
```

Comment:
yep. removed. Now any stream that is destroyed will cause the baseSystemTime entry to be removed